src/r_draw.c

```diff
@@ -136,6 +136,8 @@ byte dc_skycolor;
                 dest += linesize;                                        \
                 if ((frac += fracstep) >= heightmask)                    \
                     frac -= heightmask;                                  \
+                if (frac < 0)                                            \
```

Comment:
Do we really need this check? I'd rather have a check for a negative `scaley` in a different place than in all `DrawColumn*` functions.

Reply:
There are already plenty of checks and `while{}` loops in this code path, and I think this additional one is simple and plausible enough. As a matter of fact, we already implicitly supported negative scaling factors, and all that was missing is this simply boundaries check, which is already there to cap the `frac` value into positive bounds for positive `fracstep` values. With this simple check, we even support negative scaling for textures. No special casing for negative `scaley` values needed anywhere else.